beacon-chain/core/helpers/committee.go

```diff
@@ -397,31 +429,82 @@ func CommitteeAssignment(
 		)
 	}
 
+	requestedEpoch := SlotToEpoch(slot)
+	if requestedEpoch != currentEpoch {
+		var selectedCommittees []*CrosslinkCommittee
+		startSlot := StartSlot(wantedEpoch)
+		for slot := startSlot; slot < startSlot+params.BeaconConfig().SlotsPerEpoch; slot++ {
+			crosslinkCommittees, err := CrosslinkCommitteesAtSlot(
+				state, slot, registryChange)
+			if err != nil {
+				return []uint64{}, 0, 0, false, fmt.Errorf("could not get crosslink committee: %v", err)
+			}
+			for _, committee := range crosslinkCommittees {
+				for _, idx := range committee.Committee {
+					if idx == validatorIndex {
+						selectedCommittees = append(selectedCommittees, committee)
+					}
+
+					if len(selectedCommittees) > 0 {
+						validators := selectedCommittees[0].Committee
+						shard := selectedCommittees[0].Shard
+						firstCommitteeAtSlot := crosslinkCommittees[0].Committee
+						isProposer := firstCommitteeAtSlot[slot%
+							uint64(len(firstCommitteeAtSlot))] == validatorIndex
+						return validators, shard, slot, isProposer, nil
+					}
+				}
+			}
+		}
+	}
+
+	assignmentsLock.Lock()
+	defer assignmentsLock.Unlock()
+	if assignment, ok := validatorAssignments[validatorIndex]; ok {
```

Comment:
what is the point of this ? this block of code will never be reached . The retrieval from the cache should be at the start of the function not the end.

Reply:
Sometimes theres a request for a previous/next epoch and this case makes sure it doesn't get the data from the current epoch.

Reply:
And it's reached due to the epoch being set every epoch.

Reply:
ah k it makes sense now, the way the code is written it looks like we only request from the cache at the end of the func. I missed this 
```go 
if requestedEpoch != currentEpoch {
```